src/plone/restapi/tests/test_services_vocabularies.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+from plone import api
+from plone.app.testing import setRoles
+from plone.app.testing import SITE_OWNER_NAME
+from plone.app.testing import SITE_OWNER_PASSWORD
+from plone.app.testing import TEST_USER_ID
+from plone.restapi.testing import PLONE_RESTAPI_DX_FUNCTIONAL_TESTING
+from plone.restapi.testing import RelativeSession
+from zope.component import provideUtility
+from zope.schema.interfaces import IVocabularyFactory
+from zope.schema.vocabulary import SimpleVocabulary
+from zope.schema.vocabulary import SimpleTerm
+
+import transaction
+import unittest
+
+
+TEST_TERM_1 = SimpleTerm(42, token='token1', title=u'Title 1')
+TEST_TERM_2 = SimpleTerm(43, token='token2', title=u'Title 2')
+TEST_VOCABULARY = SimpleVocabulary([TEST_TERM_1, TEST_TERM_2])
+
+
+def test_vocabulary_factory(context):
+    return TEST_VOCABULARY
+
+
+def test_context_vocabulary_factory(context):
+    return SimpleVocabulary([
+        SimpleTerm(context.id, token='id', title=context.id),
+        SimpleTerm(context.title, token='title', title=context.title)
+        ])
+
+
+class TestVocabularyEndpoint(unittest.TestCase):
+
+    layer = PLONE_RESTAPI_DX_FUNCTIONAL_TESTING
+
+    maxDiff = None
+
+    def setUp(self):
+        self.app = self.layer['app']
+        self.portal = self.layer['portal']
+        self.portal_url = self.portal.absolute_url()
+        setRoles(self.portal, TEST_USER_ID, ['Manager'])
+
+        self.api_session = RelativeSession(self.portal_url)
+        self.api_session.headers.update({'Accept': 'application/json'})
+        self.api_session.auth = (SITE_OWNER_NAME, SITE_OWNER_PASSWORD)
+        provideUtility(test_vocabulary_factory, provides=IVocabularyFactory,
+                       name='plone.restapi.tests.test_vocabulary')
+
+    def test_get_vocabulary(self):
+        response = self.api_session.get(
+            '/@vocabularies/plone.restapi.tests.test_vocabulary')
+
+        self.assertEqual(200, response.status_code)
+        response = response.json()
+        self.assertEqual(
+            response,
+            {u'@id': u'http://localhost:55001/plone/@vocabularies/''plone.restapi.tests.test_vocabulary',  # noqa
```

Comment:
@sneridagh I refactored the dict and now see the test failure trace I get:

```
    151/153 (98.7%)

Failure in test test_get_all_vocabularies (plone.restapi.tests.test_services_vocabularies.TestVocabularyEndpoint)
Traceback (most recent call last):
  File "/usr/lib/python2.7/unittest/case.py", line 331, in run
    testMethod()
  File "/home/timo/workspace/plone/plone.restapi/src/plone/restapi/tests/test_services_vocabularies.py", line 84, in test_get_all_vocabularies
    self.assertIn('plone.app.vocabularies.Roles', response)
  File "/usr/lib/python2.7/unittest/case.py", line 805, in assertIn
    self.fail(self._formatMessage(msg, standardMsg))
  File "/usr/lib/python2.7/unittest/case.py", line 412, in fail
    raise self.failureException(msg)
AssertionError: 'plone.app.vocabularies.Roles' not found in [{u'@id': u'plone.app.vocabularies.Skins/', u'title': u'plone.app.vocabularies.Skins'}, {u'@id': u'plone.app.vocabularies.AvailableContentLanguages/', u'title': u'plone.app.vocabularies.AvailableContentLanguages'}, {u'@id': u'plone.schemaeditor.VocabulariesVocabulary/', u'title': u'plone.schemaeditor.VocabulariesVocabulary'}, {u'@id': u'plone.app.event.Weekdays/', u'title': u'plone.app.event.Weekdays'}, {u'@id': u'plone.contentrules.events/', u'title': u'plone.contentrules.events'}, {u'@id': u'plone.restapi.tests.test_vocabulary/', u'title': u'plone.restapi.tests.test_vocabulary'}, {u'@id': u'plone.app.discussion.vocabularies.CaptchaVocabulary/', u'title': u'plone.app.discussion.vocabularies.CaptchaVocabulary'}, {u'@id': u'plone.app.vocabularies.Actions/', u'title': u'plone.app.vocabularies.Actions'}, {u'@id': u'plone.app.contenttypes.migration.changed_base_classes/', u'title': u'plone.app.contenttypes.migration.changed_base_classes'}, {u'@id': u'wicked.vocabularies.BaseConfigurationsOptions/', u'title': u'wicked.vocabularies.BaseConfigurationsOptions'}, {u'@id': u'plone.app.content.ValidAddableTypes/', u'title': u'plone.app.content.ValidAddableTypes'}, {u'@id': u'plone.app.vocabularies.CommonTimezones/', u'title': u'plone.app.vocabularies.CommonTimezones'}, {u'@id': u'plone.restapi.tests.test_context_vocabulary/', u'title': u'plone.restapi.tests.test_context_vocabulary'}, {u'@id': u'plone.app.vocabularies.AvailableEditors/', u'title': u'plone.app.vocabularies.AvailableEditors'}, {u'@id': u'cmf.calendar.AvailableEventTypes/', u'title': u'cmf.calendar.AvailableEventTypes'}, {u'@id': u'plone.app.vocabularies.SupportedContentLanguages/', u'title': u'plone.app.vocabularies.SupportedContentLanguages'}, {u'@id': u'plone.app.vocabularies.Keywords/', u'title': u'plone.app.vocabularies.Keywords'}, {u'@id': u'plone.app.vocabularies.Groups/', u'title': u'plone.app.vocabularies.Groups'}, {u'@id': u'plone.app.vocabularies.AvailableTimezones/', u'title': u'plone.app.vocabularies.AvailableTimezones'}, {u'@id': u'plone.app.vocabularies.Catalog/', u'title': u'plone.app.vocabularies.Catalog'}, {u'@id': u'plone.app.discussion.vocabularies.TextTransformVocabulary/', u'title': u'plone.app.discussion.vocabularies.TextTransformVocabulary'}, {u'@id': u'wicked.vocabularies.CacheConfigurationsOptions/', u'title': u'wicked.vocabularies.CacheConfigurationsOptions'}, {u'@id': u'plone.app.vocabularies.Month/', u'title': u'plone.app.vocabularies.Month'}, {u'@id': u'plone.app.vocabularies.Weekdays/', u'title': u'plone.app.vocabularies.Weekdays'}, {u'@id': u'plone.app.vocabularies.WorkflowTransitions/', u'title': u'plone.app.vocabularies.WorkflowTransitions'}, {u'@id': u'plone.app.vocabularies.WeekdaysShort/', u'title': u'plone.app.vocabularies.WeekdaysShort'}, {u'@id': u'plone.app.contenttypes.metadatafields/', u'title': u'plone.app.contenttypes.metadatafields'}, {u'@id': u'Behaviors/', u'title': u'Behaviors'}, {u'@id': u'Interfaces/', u'title': u'Interfaces'}, {u'@id': u'plone.app.event.Timezones/', u'title': u'plone.app.event.Timezones'}, {u'@id': u'plone.app.vocabularies.WorkflowStates/', u'title': u'plone.app.vocabularies.WorkflowStates'}, {u'@id': u'plone.app.event.EventTypes/', u'title': u'plone.app.event.EventTypes'}, {u'@id': u'Fields/', u'title': u'Fields'}, {u'@id': u'plone.app.contenttypes.migration.atctypes/', u'title': u'plone.app.contenttypes.migration.atctypes'}, {u'@id': u'plone.app.vocabularies.SyndicatableFeedItems/', u'title': u'plone.app.vocabularies.SyndicatableFeedItems'}, {u'@id': u'plone.app.vocabularies.Users/', u'title': u'plone.app.vocabularies.Users'}, {u'@id': u'plone.app.event.AvailableTimezones/', u'title': u'plone.app.event.AvailableTimezones'}, {u'@id': u'plone.app.vocabularies.WeekdaysAbbr/', u'title': u'plone.app.vocabularies.WeekdaysAbbr'}, {u'@id': u'plone.app.vocabularies.UserFriendlyTypes/', u'title': u'plone.app.vocabularies.UserFriendlyTypes'}, {u'@id': u'plone.app.vocabularies.PortalTypes/', u'title': u'plone.app.vocabularies.PortalTypes'}, {u'@id': u'plone.app.controlpanel.WickedPortalTypes/', u'title': u'plone.app.controlpanel.WickedPortalTypes'}, {u'@id': u'plone.app.vocabularies.SyndicationFeedTypes/', u'title': u'plone.app.vocabularies.SyndicationFeedTypes'}, {u'@id': u'plone.app.vocabularies.Timezones/', u'title': u'plone.app.vocabularies.Timezones'}, {u'@id': u'plone.formwidget.relations.cmfcontentsearch/', u'title': u'plone.formwidget.relations.cmfcontentsearch'}, {u'@id': u'plone.app.vocabularies.ReallyUserFriendlyTypes/', u'title': u'plone.app.vocabularies.ReallyUserFriendlyTypes'}, {u'@id': u'plone.app.vocabularies.AllowableContentTypes/', u'title': u'plone.app.vocabularies.AllowableContentTypes'}, {u'@id': u'plone.app.contenttypes.migration.extendedtypes/', u'title': u'plone.app.contenttypes.migration.extendedtypes'}, {u'@id': u'plone.app.vocabularies.MonthAbbr/', u'title': u'plone.app.vocabularies.MonthAbbr'}, {u'@id': u'plone.app.vocabularies.Workflows/', u'title': u'plone.app.vocabularies.Workflows'}, {u'@id': u'Group Ids/', u'title': u'Group Ids'}, {u'@id': u'plone.app.event.SynchronizationStrategies/', u'title': u'plone.app.event.SynchronizationStrategies'}, {u'@id': u'plone.app.vocabularies.AllowedContentTypes/', u'title': u'plone.app.vocabularies.AllowedContentTypes'}, {u'@id': u'plone.app.vocabularies.Roles/', u'title': u'plone.app.vocabularies.Roles'}]

                    
  Ran 153 tests with 1 failures and 0 errors in 25.811 seconds.
```

Got my point? ;)